### PR TITLE
[CI] Add llvm-mingw

### PR DESCRIPTION
**Pull request by past-due**
Add llvm-mingw CI pass.

Depends on:
- [x] ~~https://github.com/getsentry/crashpad/pull/78~~
- [x] ~~https://github.com/getsentry/sentry-native/pull/794~~

### CODE REVIEW

**Review by supervacuus — CHANGES_REQUESTED** [review]
Thank you for this contribution @past-due!

Please let me also be clear about what this CI addition means in terms of maintenance responsibility:
* the responsibility to fix issues concerning MinGW crashpad support still lies with the community
* if the build fails, we will create an issue tagging whoever is interested (I would certainly tag you @past-due)
* if no one is found who has the time to fix the issue, we will deactivate the build so as not to block progress on topics orthogonal to MinGW
* if no one fixed the issue until the next release, the changelog will contain a note that the MinGW support for crashpad is broken (referencing the tracking issue)

@Swatinem, let us finish the remaining crashpad updates in a separate PR and remove the submodule update here.

**Comment by supervacuus** [comment]
CC: @bitsandfoxes @markushi

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -135,6 +141,64 @@ jobs:
...
         if: ${{ runner.os == 'macOS' }}
         run: echo $(brew --prefix llvm@15)/bin >> $GITHUB_PATH
+      
+      - name: Installing LLVM-MINGW Dependencies
+        if: ${{ runner.os == 'Windows' && env['TEST_MINGW'] }}
+        shell: powershell
+        env:
+          MINGW_PKG_PREFIX: ${{ matrix.MINGW_PKG_PREFIX }}
+          MINGW_ASM_MASM_COMPILER: ${{ matrix.MINGW_ASM_MASM_COMPILER }}
+        run: |
```
[the remark this answers is not in the document]
Comment (past-due) [reply .github/workflows/ci.yml]: Done 👍

**Comment by past-due** [comment]
> Please let me also be clear about what this CI addition means in terms of maintenance responsibility:
> 
> * the responsibility to fix issues concerning MinGW crashpad support still lies with the community
> * if the build fails, we will create an issue tagging whoever is interested (I would certainly tag you @past-due)
> * if no one is found who has the time to fix the issue, we will deactivate the build so as not to block progress on topics orthogonal to MinGW
> * if no one fixed the issue until the next release, the changelog will contain a note that the MinGW support for crashpad is broken (referencing the tracking issue)
> 

Sounds great. 👍

Please feel free to tag me if / when any future breakages happen.

**Review by supervacuus — APPROVED** [review]

**Pull request merged**